sql/src/test/java/io/crate/metadata/SchemasITest.java

```diff
@@ -180,4 +190,99 @@ public void testClusterTable() throws Exception {
             clusterService.state(), routingProvider, null, null, SessionContext.systemSessionContext()
         ).locations().size(), is(1));
     }
+
+    @Test
+    public void testResolveTableInfoForValidFQN() throws IOException {
+        RelationName tableIdent = new RelationName("schema", "t");
+        SQLExecutor sqlExecutor = getSqlExecutorBuilderForTable(tableIdent, "doc", "schema").build();
+
+        QualifiedName fqn = QualifiedName.of("schema", "t");
+        TableInfo tableInfo = sqlExecutor.schemas()
+            .resolveTableInfo(fqn, Operation.READ, sqlExecutor.getSessionContext().searchPath());
+
+        RelationName relation = tableInfo.ident();
+        assertThat(relation.schema(), Matchers.is("schema"));
+        assertThat(relation.name(), Matchers.is("t"));
+    }
+
+    private SQLExecutor.Builder getSqlExecutorBuilderForTable(RelationName tableIdent, String... searchPath) {
+        return SQLExecutor.builder(clusterService())
+            .addDocTable(TestingTableInfo.builder(tableIdent, shardRouting(tableIdent.name()))
```

Comment:
Minor nitpick - could we switch to the newer `addTable(createTableStatement)` here? The `addDocTable` variant has the problem that the shard-routing doesn't end up in the clusterState and therefore the tests could be based on an "out of sync" representation. We should eventually replace it completely. (And probably mark it as deprecated until then)